sdk/storage/azure-storage-common/inc/azure/storage/common/storage_common.hpp

```diff
@@ -64,7 +50,25 @@ namespace Azure { namespace Storage {
     /**
      * @brief The algorithm used for hash.
      */
-    HashAlgorithm Algorithm;
+    HashAlgorithm Algorithm = HashAlgorithm::Md5;
   };
 
+
+  namespace Details {
+    struct CaseInsensitiveComparator
+    {
+      bool operator()(const std::string& lhs, const std::string& rhs) const
+      {
+        return std::lexicographical_compare(
+            lhs.begin(), lhs.end(), rhs.begin(), rhs.end(), [](char c1, char c2) {
+              return Core::Strings::ToLower(c1) < Core::Strings::ToLower(c2);
+            });
+      }
+    };
+
+    ContentHash FromBase64String(const std::string& base64String, HashAlgorithm algorithm);
+    std::string ToBase64String(const ContentHash& hash);
```

Comment:
The interface of these two functions are not symmetrical, but their names are symmetrical.

In the first function you have input (base64 encoded string, algorithm) and the output is `ContentHash`.
But in the second one, the output is just base64 encoded string, the algorithm information is lost.

Reply:
How about `FromBase64StringHash`? I don't have good namings in mind.

Reply:
Will remain as is before we had a good name.